src/info/info_client.rs

```diff
@@ -139,6 +161,22 @@ impl InfoClient {
         serde_json::from_str(&return_data).map_err(|e| Error::JsonParse(e.to_string()))
     }
 
+    pub async fn user_token_balances(&self, address: H160) -> Result<UserTokenBalanceResponse> {
+        let input = InfoRequest::UserTokenBalances { user: address };
```

Comment:
a lot of this code is copy-paste, can you define an inner, non-pub function that takes in `InfoRequest` along with `address` and returns a `Result<T>`?

Reply:
Done